Bio/bgzf.py

```diff
@@ -446,7 +445,8 @@ def _load_bgzf_block(handle, text_mode=False):
     if expected_crc != crc:
         raise RuntimeError("CRC is %s, not %s" % (crc, expected_crc))
     if text_mode:
-        return block_size, _as_string(data)
+        # Note ISO-8859-1 aka Latin-1 preserves first 256 chars
+        return block_size, codecs.latin_1_decode(data)[0]
```

Comment:
Isn't it inconsistent to use Latin-1 here, but UTF-8 in the `write` function below?
For example,
```python
>>> import codecs
>>> s = "ü"
>>> b = s.encode()
>>> codecs.latin_1_decode(b)
('Ã¼', 2)
>>> codecs.utf_8_decode(b)
('ü', 2)
```
I would also suggest to use `decode` on the string directly, as in
```python
>>> b.decode('UTF8')
'ü'
>>> b.decode('latin-1')
'Ã¼'
```
Then you don't have to extract just the first value from the tuple returned by `codecs.utf_8_decode`, `codecs.latin_1_decode`. (If it's UTF-8, you can just do `b.decode()`).

Reply:
Given Andrey isn't available to further improve this pull request right now (and his work as drawn attention to this pre-existing issue rather than caused it), I will merge this and then follow up on ``Bio/bgzf.py`` separately.

Using codecs does seem redundant here.